aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cpp

```diff
@@ -322,14 +322,11 @@ bool is_safe_to_get_storage_as_tensor(const NestedTensorImpl* tensor) {
   const int64_t* tensor_size_ptr = tensor_sizes.data_ptr<int64_t>();
   const int64_t* tensor_stride_ptr = tensor_strides.data_ptr<int64_t>();
 
-  int64_t offset_constant = (tensor_offsets[1] - tensor_offsets[0]) /
-      tensor_size_ptr[0] * tensor_stride_ptr[0];
-
+  int64_t offset_constant = (tensor_offsets[1] - tensor_offsets[0]) / (tensor_size_ptr[0] * tensor_stride_ptr[0]);
   for (int64_t i = 2; i < n_tensors; i++) {
-    int64_t current_offset_constant =
-        (tensor_offsets[i] - tensor_offsets[i - 1]) /
-        tensor_size_ptr[(i - 1) * tensor_stride_0] *
-        tensor_stride_ptr[(i - 1) * tensor_stride_0];
+    // TODO: When 0 seq_len nested tensors are allowed we need to guard against this
+    int64_t previous_numel = tensor_size_ptr[(i - 1) * tensor_stride_0] * tensor_stride_ptr[(i - 1) * tensor_stride_0];
+    int64_t current_offset_constant = (tensor_offsets[i] - tensor_offsets[i - 1]) / previous_numel;
```

Comment:
Should we add a check that `(tensor_offsets[i] - tensor_offsets[i - 1]) % previous_numel == 0` before dividing here - and also before the loop?

Reply:
https://github.com/drisspg/pytorch/blob/029ab31c0602683821ebb6d9ce78be1fa70770f7/aten/src/ATen/native/transformers/cuda/sdp_utils.h#L65

Yes we very much do but spoke to christian and this PR is essentially neuter the SDP inclusion in native mha and then in a follow up PR expand scope